src/Elliptic/DiscontinuousGalerkin/NumericalFluxes/FirstOrderInternalPenalty.hpp

```diff
@@ -0,0 +1,379 @@
+                 NormalDotNormalDotFlux<AuxiliaryFieldTags>...>;
+
+  // Following the packaged_data pointer, this function expects as arguments the
+  // types in `argument_tags`.
+  void package_data(
+      const gsl::not_null<Variables<package_tags>*> packaged_data,
+      const db::item_type<::Tags::NormalDotFlux<
+          AuxiliaryFieldTags>>&... normal_dot_auxiliary_field_fluxes,
+      const db::item_type<::Tags::div<
+          ::Tags::Flux<AuxiliaryFieldTags, tmpl::size_t<Dim>,
+                       Frame::Inertial>>>&... div_auxiliary_field_fluxes,
+      const FluxesComputer& fluxes_computer,
+      const db::item_type<FluxesArgs>&... fluxes_args,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& interface_unit_normal)
+      const noexcept {
+    auto principal_div_aux_field_fluxes = make_with_value<Variables<tmpl::list<
+        ::Tags::Flux<FieldTags, tmpl::size_t<Dim>, Frame::Inertial>...>>>(
+        interface_unit_normal, 0.);
+    auto principal_ndot_aux_field_fluxes = make_with_value<Variables<tmpl::list<
+        ::Tags::Flux<FieldTags, tmpl::size_t<Dim>, Frame::Inertial>...>>>(
+        interface_unit_normal, 0.);
+    fluxes_computer.apply(
+        make_not_null(
+            &get<::Tags::Flux<FieldTags, tmpl::size_t<Dim>, Frame::Inertial>>(
+                principal_div_aux_field_fluxes))...,
+        fluxes_args..., div_auxiliary_field_fluxes...);
+    fluxes_computer.apply(
+        make_not_null(
+            &get<::Tags::Flux<FieldTags, tmpl::size_t<Dim>, Frame::Inertial>>(
+                principal_ndot_aux_field_fluxes))...,
+        fluxes_args..., normal_dot_auxiliary_field_fluxes...);
+    const auto helper =
+        [
+          &packaged_data, &principal_div_aux_field_fluxes,
+          &principal_ndot_aux_field_fluxes, &interface_unit_normal
+        ](const auto field_tag_v, const auto auxiliary_field_tag_v,
+          const auto normal_dot_auxiliary_field_flux) noexcept {
+      using field_tag = std::decay_t<decltype(field_tag_v)>;
+      using auxiliary_field_tag = std::decay_t<decltype(auxiliary_field_tag_v)>;
+      // Compute n.F_v(u)
+      get<::Tags::NormalDotFlux<auxiliary_field_tag>>(*packaged_data) =
+          normal_dot_auxiliary_field_flux;
+      // Compute n.F_u(div(F_v(u))) and n.F_u(n.F_v(u))
+      normal_dot_flux(
+          make_not_null(
+              &get<NormalDotDivFlux<auxiliary_field_tag>>(*packaged_data)),
+          interface_unit_normal,
+          get<::Tags::Flux<field_tag, tmpl::size_t<Dim>, Frame::Inertial>>(
+              principal_div_aux_field_fluxes));
+      normal_dot_flux(
+          make_not_null(&get<NormalDotNormalDotFlux<auxiliary_field_tag>>(
+              *packaged_data)),
+          interface_unit_normal,
+          get<::Tags::Flux<field_tag, tmpl::size_t<Dim>, Frame::Inertial>>(
+              principal_ndot_aux_field_fluxes));
+    };
+    EXPAND_PACK_LEFT_TO_RIGHT(helper(FieldTags{}, AuxiliaryFieldTags{},
```

Comment:
Does EXPAND_PACK_LEFT_TO_RIGHT plug in each Field Tag, AuxFieldTag, and normal_dot_auxiliary-field_flux to helper(), calling helper() multiple times, to load up the packaged data? If not, can you please explain how this works? Also, why do FieldTags{} and AuxiliaryFieldTags{} have braces but not normal_dot_auxiliary_field_fluxes?

Reply:
That is indeed what it does. See https://spectre-code.org/group__utilitiesgroup#ga902d1cefbb10846eb8f5be9f6be5c340 for the documentation of this macro.

Reply:
Thanks! I've mostly got it, but I still don't get why FieldTags{} and AuxiliaryFieldTags{} have braces but not normal_dot_auxiliary_field_fluxes...can you please clue me in?

Reply:
`FieldTags` is a parameter pack and `normal_dot_auxiliary_field_fluxes` are variadic arguments, so these are different language concepts. I have not tried to understand how the `EXPAND_PACK` macro handles them internally :)